wallet-new/integration/TransactionSpecs.hs

```diff
@@ -18,19 +19,29 @@ import           Test.QuickCheck (arbitrary, generate)
 import           Util
 
 
-transactionSpecs :: WalletRef -> WalletClient IO -> Spec
-transactionSpecs wRef wc = do
-    describe "Transactions" $ do
-        it "Posted transactions appear in the index" $ do
-            genesis <- makeGenesisWallet wc
-            (fromAcct, _) <- getFirstAccountAndAddress wc genesis
+transactionSpecs :: HasCallStack => WalletClient IO -> Spec
+transactionSpecs wc = do
+    it "Estimate fees of a well-formed transaction" $ estimateTransactionFees wc
+    before (createRandomSampleWallet wc) $
+         it "Fails if you spend too much money" $ spendTooMuchMoney wc
+    before ((,) <$> lookupGenesisWallet wc <*> createRandomSampleWallet wc) $
+             it "Posted transactions appear in the index" $ testPostTransaction wc
+    before (lookupGenesisWallet wc) $
+       xit "Payment: genesis wallet -> external wallet -> external wallet" $ \genesisWallet -> do
+            makeExternalDestWallet wc >>= payFromExternalWalletToExternal wc genesisWallet
 
-            wallet <- sampleWallet wRef wc
-            (toAcct, toAddr) <- getFirstAccountAndAddress wc wallet
+    before (lookupGenesisWallet wc) $
+       xit "Payment: genesis wallet -> external wallet -> regular wallet" $ \genesisWallet -> do
+            makeRegularDestWallet wc >>= payFromExternalWalletToExternal wc genesisWallet
```

Comment:
To be honest, I am not a fan of breaking the spec up into a ton of little functions like this. It makes it more difficult for me to read the specs and get a clear idea for what's going on, because I must jump up and down the file to figure out what is happening. With the old format, everything was laid out in front of you.

Could you list the benefits you find from this structure?